File: cache/src/main/java/com/mm/http/CacheRequest.java
package com.mm.http;

import com.mm.http.cache.StrategyType;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.TimeUnit;

/**
 * Created by : majian
 * Date : 2021/8/24
 */

public class CacheRequest {
    private final int cacheStrategy;
    private final Long duration;
    private final TimeUnit timeUnit;
    private final List<String> ignoreKey;

    private CacheRequest(int cacheStrategy, long duration, TimeUnit timeUnit, List<String> ignoreKey) {
        this.cacheStrategy = cacheStrategy;
        this.duration = duration;
        this.timeUnit = timeUnit;
        this.ignoreKey = ignoreKey;
    }

    public int getCacheStrategy() {
        return cacheStrategy;
    }

    public Long getDuration() {
        return duration;
    }

    public TimeUnit getTimeUnit() {
        return timeUnit;
    }

    public List<String> getIgnoreKey() {
        return ignoreKey;
    }

    public static class Builder {
        private int strategy = StrategyType.NO_CACHE;
        private Long duration = 24L;
        private TimeUnit timeUnit = TimeUnit.HOURS;
        private List<String> ignoreKey = new ArrayList<>();

        Builder cacheStrategy(@StrategyType int cacheStrategy) {
            this.strategy = cacheStrategy;
            return this;
        }

        Builder duration(Long duration) {
            this.duration = duration;
            return this;
        }

        Builder timeUnit(TimeUnit timeUnit) {
            this.timeUnit = timeUnit;
            return this;
        }

        Builder ignoreKey(List<String> ignoreKey) {
            this.ignoreKey = ignoreKey;
            return this;
        }

        CacheRequest build() {
            return new CacheRequest(strategy, duration, timeUnit, ignoreKey);
        }
    }
}

File: retrofit_cache/src/main/java/com/mm/http/CacheRequest.kt
package com.mm.http

import com.mm.http.cache.StrategyType
import java.util.ArrayList
import java.util.concurrent.TimeUnit

/**
 * 缓存请求参数
 */
class CacheRequest private constructor(
    val cacheStrategy: Int,
    val duration: Long,
    val timeUnit: TimeUnit,
    val ignoreKey: List<String>
) {

    class Builder {
        private var strategy = StrategyType.NO_CACHE
        private var duration = 24L
        private var timeUnit = TimeUnit.HOURS
        private var ignoreKey: List<String> = ArrayList()
        fun cacheStrategy(@StrategyType cacheStrategy: Int): Builder {
            strategy = cacheStrategy
            return this
        }

        fun duration(duration: Long): Builder {
            this.duration = duration
            return this
        }

        fun timeUnit(timeUnit: TimeUnit): Builder {
            this.timeUnit = timeUnit
            return this
        }

        fun ignoreKey(ignoreKey: List<String>): Builder {
            this.ignoreKey = ignoreKey
            return this
        }

        fun build(): CacheRequest {
            return CacheRequest(strategy, duration, timeUnit, ignoreKey)
        }
    }
}